### PR TITLE
[ARM] Create mapping symbols with non-unique names

### PR DESCRIPTION
**Pull request by MaskRay**
Similar to #99836 for AArch64.

Non-unique names save .strtab space and match GNU assembler.

### CODE REVIEW

File: llvm/lib/Target/ARM/MCTargetDesc/ARMELFStreamer.cpp [hunk 1]
```diff
@@ -670,17 +670,15 @@ class ARMELFStreamer : public MCELFStreamer {
   }
 
   void EmitMappingSymbol(StringRef Name) {
-    auto *Symbol = cast<MCSymbolELF>(getContext().getOrCreateSymbol(
-        Name + "." + Twine(MappingSymbolCounter++)));
+    auto *Symbol = cast<MCSymbolELF>(getContext().createLocalSymbol(Name));
```
Comment (smithp35) [inline llvm/lib/Target/ARM/MCTargetDesc/ARMELFStreamer.cpp:673]: I think these two functions are the only use of MappingSymbolCounter (similar to AArch64). Can it be removed?

**Review by smithp35 — COMMENTED** [review]
As with AArch64 I think MappingSymbolCounter can be removed as unused. Other than that looks good to me.

File: llvm/lib/Target/ARM/MCTargetDesc/ARMELFStreamer.cpp (hunk 1 above)
Comment (MaskRay) [reply llvm/lib/Target/ARM/MCTargetDesc/ARMELFStreamer.cpp:673]: Removed!

**Review by smithp35 — APPROVED** [review]
LGTM thanks for the update.

**Pull request merged**